Zend/zend_alloc.c

```diff
@@ -1730,10 +1730,15 @@ static void *zend_mm_alloc_huge(zend_mm_heap *heap, size_t size ZEND_FILE_LINE_D
 	void *ptr;
 
 #if ZEND_MM_LIMIT
+	if (UNEXPECTED(new_size == 0)) {
+		/* overflow in ZEND_MM_ALIGNED_SIZE_EX */
+		goto memory_limit_exhausted;
+	}
 	if (UNEXPECTED(new_size > heap->limit - heap->real_size)) {
 		if (zend_mm_gc(heap) && new_size <= heap->limit - heap->real_size) {
 			/* pass */
 		} else if (heap->overflow == 0) {
+			memory_limit_exhausted:
```

Comment:
By convention, the goto label should not be indented.

Reply:
Thanks! Wasn't sure.